packages/core/src/transport/transport.ts

```diff
@@ -29,7 +51,32 @@ export class HttpRequest {
         reportBeaconError(e)
       }
     }
+
+    const transportIntrospection = (event: ProgressEvent) => {
+      const req = event?.currentTarget as XMLHttpRequest
```

Comment:
`transportIntrospection` is called unconditionally during `loadend`, even if the request did succeed. You should add a check here to skip sending the failure message if it succeeded (check on request status and maybe response text